sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -161,20 +156,14 @@ static class Factory<InputT, RestrictionT, PositionT, WatermarkEstimatorStateT,
       implements PTransformRunnerFactory {
 
     @Override
-    public final void addRunnerForPTransform(Context context) {
+    public final void addRunnerForPTransform(Context context) throws IOException {
```

Comment:
This changed method signature which is breaking change. Is FnApiDoFnRunner purely internal?

Reply:
Yes, all this code is purely internal. The entire module sdks/_<language>_/harness is internal.